router/router.go

```diff
@@ -33,13 +41,59 @@ func Create(db *database.GormDatabase, vInfo *model.VersionInfo, conf *config.Co
 		DB:       db,
 		ImageDir: conf.UploadedImagesDir,
 	}
-	userHandler := api.UserAPI{DB: db, PasswordStrength: conf.PassStrength, NotifyDeleted: streamHandler.NotifyDeletedUser}
+	userChangeNotifier := new(api.UserChangeNotifier)
+	userHandler := api.UserAPI{DB: db, PasswordStrength: conf.PassStrength, UserChangeNotifier: userChangeNotifier}
+	userChangeNotifier.OnUserDeleted(streamHandler.NotifyDeletedUser)
+	pluginHandler := api.PluginAPI{
+		Notifier: streamHandler,
+		DB:       db,
+		Holder:   holder.NewPluginInstanceHolder(),
+	}
+	userChangeNotifier.OnUserDeleted(pluginHandler.DestroyPluginsForUser)
+	userChangeNotifier.OnUserAdded(pluginHandler.InitPluginsForUser)
 
 	g := gin.New()
 
 	g.Use(gin.Logger(), gin.Recovery(), error.Handler(), location.Default())
 	g.NoRoute(error.NotFound())
 
+	{
+		g.GET("/plugin", authentication.RequireClient(), pluginHandler.GetPlugins)
+		pluginRoute := g.Group("/plugin/", authentication.RequirePlugin())
```

Comment:
Yeah, the endpoint design might need some changes. Actually, I hit a problem while designing custom routers for plugins. Since you need to provide a `RouterGroup` to the plugin, it is inevitable to register specific routes such as `/plugin/Pabcd/custom/`. However I am not sure whether it is a [limitation](https://github.com/gin-gonic/gin/issues/1301) or the `gin` team [disproves](https://github.com/gin-gonic/gin/issues/1681) it explicitly, a wildcard route and a specific route cannot co-exist even though they don't actually overlap(eg: `/plugin/:id/enable` and `/plugin/Pabcd/custom/`).

Reply:
This is resolved too right?